src/codegen/c.zig

```diff
@@ -839,10 +839,13 @@ pub const DeclGen = struct {
         try w.writeAll("(");
         const param_len = dg.decl.ty.fnParamLen();
 
+        const target = dg.module.getTarget();
         var index: usize = 0;
         var params_written: usize = 0;
         while (index < param_len) : (index += 1) {
-            if (dg.decl.ty.fnParamType(index).zigTypeTag() == .Void) continue;
+            const param_type = dg.decl.ty.fnParamType(index);
+            if (param_type.zigTypeTag() == .Void) continue;
+            if (param_type.isInt() and param_type.intInfo(target).bits == 0) continue;
```

Comment:
Actually this fails to output the correct number of function arguments in certain cases -- tests fail with:

```
test/zig-cache/o/71ea89031adc7a84a0790b90c485965a/test.c:12256:29: error: too many arguments to function call, expected 0, have 1
276test/zig-cache/o/71ea89031adc7a84a0790b90c485965a/test.c:2420:13: note: 'behavior_enum_doALoopThing' declared here
```

which corresponds to: https://github.com/ziglang/zig/blob/cb3b1dd6ddee65d1811fb4058b5cc0f2c06d0139/test/behavior/enum.zig#L854

and fails because enum literals don't have comptime bits: https://github.com/ziglang/zig/blob/cb3b1dd6ddee65d1811fb4058b5cc0f2c06d0139/src/type.zig#L2050-L2072

so they also trigger the `continue` statement here.

Should I restore the original version, or do you have an alternative suggestion?

Reply:
It's correct to also omit the parameter for enums with only 1 tag. Likely the code that implements the function type signature also needs to be adjusted from explicitly checking `void` to checking `hasRuntimeBitsIgnoreComptime`.

Indeed, looking at `renderPtrToFnTypedef` I see similar logic that needs to be updated:

```zig
            if (fn_info.param_types[index].zigTypeTag() == .Void) continue;
```

Reply:
Thanks for the catch, I updated `renderPtrToFnTypedef()` to match.

The issue I was encountering was that `airCall()` also lacked this logic, and would emit certain parameters for types that have no runtime bits in the generated function call. Tests seem to be working now.